### PR TITLE
Write celer-sim results as 'null' when they're disabled

### PR DESCRIPTION
**Pull request by sethrj**
This writes `null` for track counters and step times if they're disabled. It also removes the call to `std::chrono::high_resolution_clock::now()` when step timers are disabled (see #1172).

### CODE REVIEW

File: app/celer-sim/simple-driver.py [hunk 1]
```diff
@@ -141,6 +142,12 @@ def strtobool(text):
     json.dump(j, f, indent=1)
 print("Results written to", outfilename, file=stderr)
 
-time = j['result']['runner']['time'].copy()
-time.pop('steps')
+run_output =j['result']['runner']
+time = run_output['time'].copy()
+steps = time.pop('steps')
+if use_device:
+    assert(len(steps) == run_output.num_steps[0])
```
Comment (amandalund) [inline app/celer-sim/simple-driver.py]:
```suggestion
    assert(len(steps[0]) == run_output['num_step_iterations'][0])
```

**Review by amandalund — APPROVED** [review]
Looks good @sethrj!

File: app/celer-sim/simple-driver.py (hunk 1 above)
Comment (sethrj) [reply app/celer-sim/simple-driver.py]: Clearly I'm too used to the Jenkins job failing...

**Comment by sethrj** [comment]
https://cloud.cees.ornl.gov/jenkins-ci/job/celeritas/job/PR-1223/lastBuild/testReport/

File: app/celer-sim/simple-driver.py [hunk 2]
```diff
@@ -141,6 +142,12 @@ def strtobool(text):
     json.dump(j, f, indent=1)
 print("Results written to", outfilename, file=stderr)
 
-time = j['result']['runner']['time'].copy()
-time.pop('steps')
+run_output =j['result']['runner']
+time = run_output['time'].copy()
+steps = time.pop('steps')
+if use_device:
+    assert(len(steps) == run_output['num_step_iterations'][0])
```
Comment (amandalund) [inline app/celer-sim/simple-driver.py:149]: Shouldn't it be `len(steps[0])` since there's an array of step times for each event?

Comment (sethrj) [reply app/celer-sim/simple-driver.py:149]: Damn it why didn't I just hit "commit suggestion" 😩

**Pull request merged**